server/manageuser.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+require_once 'config.php';
+require_once "user.class.php";
+
+$USER = new User();
+
+if (!$USER->hasPermission('edit_user_permission')) {
+	header("Location: /unauthorized");
+	die();
+}
+
+if(isset($_REQUEST['callback'])){
+	switch ($_REQUEST['callback']) {
+		case 'getUserTable':
+			getUserTable($_REQUEST);
+			break;
+		case 'updateUserPermissions':
+			updateUserPermissions($_REQUEST);
+			break;
+		case 'addUser':
+			addUser($_REQUEST);
+			break;
+		default:
+			die('Unregistered command. This instance has been reported.');
+			break;
+	}
+}
+
+function getUserTable($data){
+	GLOBAL $DB_CONN;
+
+	$response = array();
+	$response['success'] = true;
+
+	$stmt = $DB_CONN->prepare("SELECT userId, firstName, lastName, email, preparesTaxes, archived 
+		FROM vita.user
+		ORDER BY firstName, lastName");
+
+	$stmt->execute(array());
+
+	$thead = "<thead><tr><th>Name</th><th>Email</th><th>Permissions</th><th>Count</th></tr>";
+	$tbody = "<tbody>";
+	while($row = $stmt->fetch(PDO::FETCH_ASSOC)){
+		$optionList = getUserPermissionOptionList($row['userId']);
```

Comment:
Also not sure how I feel about executing another SQL request for each user... Why not combine all of this into something (albeit a little uglier, but also better if you go the AJAX data / front-end table creation method, see comment below).
```
SELECT User.userId, firstName, lastName, email, preparesTaxes, User.archived, Permission.permissionId, Permission.name, userPermissionId 
FROM User
JOIN UserPermission ON User.userId = UserPermission.userId
JOIN Permission ON UserPermission.permissionId = Permission.permissionId
WHERE User.userId = ?
```

Reply:
You would also need to select out a list of all permissions and then somehow workout the logic to attach unused permissions to each user. Or rather change how it's designed on the frontend. I wouldn't worry about performance on this page due to how often it will be used in practice.

Reply:
Hm I think you're right about not worrying about this too much, but this is definitely a place to start optimizing if we ever needed to.